File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ScopeGrantedEvent.java
package work.socialhub.kslack.entity.event;

import java.util.List;

/**
 *
 */
public class ScopeGrantedEvent implements Event {

    public static final String TYPE_NAME = "scope_granted";

    private final String type = TYPE_NAME;
    private List<String> scopes;
    private String triggerId;

    public String getType() {
        return this.type;
    }

    public List<String> getScopes() {
        return this.scopes;
    }

    public String getTriggerId() {
        return this.triggerId;
    }

    public void setScopes(List<String> scopes) {
        this.scopes = scopes;
    }

    public void setTriggerId(String triggerId) {
        this.triggerId = triggerId;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ScopeGrantedEvent.kt
package work.socialhub.kslack.entity.event

/**
 *
 */
class ScopeGrantedEvent : Event {
    override val type: String = TYPE_NAME
    var scopes: Array<String>? = null
    var triggerId: String? = null

    companion object {
        const val TYPE_NAME: String = "scope_granted"
    }
}